uwblueprint/paramedics-react
# RTU stage3 - integrate online-offline detection on CCP Page

[Notion ticket](https://www.notion.so/uwblueprintexecs/9224caeb21ce4350b8a1dc378d393323?v=43f44ac04d734394b7f7edc636c523b4)

[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-(S20%2C-F20)?node-id=1972%3A36028)

## Changes
- Refactored apollo client to use a subscription client to detect connected, disconnected, and reconnecting events
- text indicator on top right should show status of network detections on ccp pages
- *DO NOT MERGE THIS BRANCH until jl-rtu-stage2 has been merged
## Screenshots
![image](https://user-images.githubusercontent.com/44094521/100524795-1951bd80-3189-11eb-8daf-737b2fa55e66.png)
![image](https://user-images.githubusercontent.com/44094521/100524798-21116200-3189-11eb-9952-43b0b5caa9ff.png)

## Testing
- start your paramedics-web backend. check that the indicator says "connected"
- then, kill the server using pmc-d. Check that the indicator says "disconnected"
- restart your server using pmc-u. the indicator should briefly say "reconnecting...", then "connected" within 15-30 seconds. This may take a while

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
